Applications/spire/source/charting/charting_window.cpp

```diff
@@ -140,6 +142,10 @@ void ChartingWindow::set_model(std::shared_ptr<ChartModel> model) {
   m_chart = new ChartView(*m_model, this);
   m_security_widget->set_widget(m_chart);
   m_chart->installEventFilter(this);
+  m_auto_scale_button->connect_clicked_signal([=] {
+    m_is_chart_auto_scaled = !m_is_chart_auto_scaled;
+    m_chart->set_auto_scale(m_is_chart_auto_scaled);
```

Comment:
Define a method and call that method.

Reply:
Did you say before that we define methods for all connect_x() functions?

Reply:
I don't think I specifically said that before. It's not strictly about the length, it's about readability. Code in the constructor should focus only on getting things setup. How things work should be kept out of it.